core/new-gui/src/app/workspace/component/result-panel/result-panel.component.ts

```diff
@@ -72,16 +72,37 @@ export class ResultPanelComponent implements OnInit {
         if (event.current.state === ExecutionState.Aborted) {
           this.resultPanelToggleService.openResultPanel();
         }
-        if (event.current.state === ExecutionState.Completed || event.current.state === ExecutionState.Running) {
-          const sinkOperators = this.workflowActionService
+        if (event.previous.state === ExecutionState.Running && event.current.state === ExecutionState.Completed) {
+          const activeSinkOperators = this.workflowActionService
             .getTexeraGraph()
             .getAllOperators()
-            .filter(op => op.operatorType.toLowerCase().includes("sink"));
-          if (sinkOperators.length > 0 && !this.currentOperatorId) {
-            this.workflowActionService.getJointGraphWrapper().unhighlightOperators(...currentlyHighlighted);
-            this.workflowActionService.getJointGraphWrapper().highlightOperators(sinkOperators[0].operatorID);
+            .filter(op => isSink(op))
+            .filter(op => !op.isDisabled)
+            .map(op => op.operatorID);
+
+          if (activeSinkOperators.length > 0) {
+            if (!(currentlyHighlighted.length == 1 && activeSinkOperators.includes(currentlyHighlighted[0]))) {
+              this.workflowActionService.getJointGraphWrapper().unhighlightOperators(...currentlyHighlighted);
+              this.workflowActionService.getJointGraphWrapper().highlightOperators(activeSinkOperators[0]);
+            }
+            this.resultPanelToggleService.openResultPanel();
+          }
+        }
```

Comment:
this is under the condition `previous==running && current==completed`. I think python messages and error would require Python UDF to be auto-highlighted and result panel auto-opened DURING execution... when completed, it is more natural to show results instead.

Reply:
I think you misread the condition, 
`previous==running && current==completed` is only for sinks, so results are shown when workflow is completed (as you said above)
`current==running` is for python UDFs, currently one python UDF operator will be highlighted as soon as workflow starts running, I understand that you wants the python to be highlighted only when messages come, but let's change that later

Reply:
Sounds good thanks